components/AdminScoringForm.tsx

```diff
@@ -0,0 +1,231 @@
+import { NextActionEnum } from '@/lib/types'
+import { Applicant, QualificationType16, QualificationType18 } from '@prisma/client'
+import { CrossCircledIcon } from '@radix-ui/react-icons'
+import { Button, Callout, Dialog, Flex, Heading, Spinner, Text, TextField } from '@radix-ui/themes'
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { format } from 'date-fns'
+import React, { FC, useEffect, useState } from 'react'
+import { useFormState } from 'react-dom'
+
+interface AdminScoringFormProps {
+  applicant: Pick<Applicant, 'cid' | 'ucasNumber' | 'firstName' | 'surname'>
+  applicationId: number
+  nextAction: NextActionEnum
+}
+
+const AdminScoringForm: FC<AdminScoringFormProps> = ({
+  applicant,
+  applicationId,
+  nextAction
+}: AdminScoringFormProps) => {
+  const [isDialogOpen, setIsDialogOpen] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+  const [adminScoringData, setAdminScoringData] = useState<AdminScoringData | null>(null)
+
+  useQuery({
+    queryKey: [applicationId],
+    queryFn: async () => {
+      const res = await fetch(`/api/adminScoringForm/${applicationId}`)
+      const data: AdminScoringData = await res.json()
+      setAdminScoringData(data)
+      return data
+    },
+    enabled: isDialogOpen
+  })
+
+  const queryClient = useQueryClient()
+
+  const upsertAdminScoringWithId = async (prevState: FormPassbackState, formData: FormData) => {
+    const res = await upsertAdminScoring(nextAction, applicationId, prevState, formData)
+    await queryClient.invalidateQueries({ queryKey: [applicationId] })
+    return res
+  }
+
+  const [state, formAction] = useFormState(upsertAdminScoringWithId, { status: '', message: '' })
+
+  useEffect(() => {
+    console.log('set loading to false')
+    setIsLoading(false)
+    if (state.status === 'success') setIsDialogOpen(false)
+  }, [state, setIsDialogOpen, setIsLoading])
+
+  return (
+    <Dialog.Root open={isDialogOpen} onOpenChange={setIsDialogOpen} defaultOpen={false}>
+      <Dialog.Trigger disabled={nextAction < NextActionEnum.ADMIN_SCORING}>
+        <Button>Admin Scoring Form</Button>
+      </Dialog.Trigger>
+
+      <Dialog.Content maxWidth="450px">
+        <Dialog.Title>Admin Scoring Form</Dialog.Title>
+
+        {adminScoringData?.imperialReview?.examRatingDone &&
+          adminScoringData?.imperialReview?.examRatingBy && (
+            <Text size="2" className="italic text-gray-500">
+              Last edited by {adminScoringData?.imperialReview?.examRatingBy} on{' '}
+              {format(adminScoringData?.imperialReview?.examRatingDone, "dd/MM/yy 'at' hh:mm")}
+            </Text>
+          )}
+        <Callout.Root className="my-5">
+          <Callout.Text size="4">
+            Applicant: {applicant.firstName} {applicant.surname}
+          </Callout.Text>
+          <Callout.Text size="4">UCAS number: {applicant.ucasNumber}</Callout.Text>
+        </Callout.Root>
```

Comment:
Should this make use of the [DataList](https://www.radix-ui.com/themes/docs/components/data-list) component instead? May look more appropriate (or not: try!)